src/components/Registers.js

```diff
@@ -1,20 +1,33 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-// import Category from './Category';
+import React, { useState } from 'react';
+// import PropTypes from 'prop-types';
 import RegMenu from './RegMenu';
-// import {
-//   Pointers,
-//   Temporary,
-//   CalleeSaved,
-//   Arguments,
-// } from '../utils/constants';
+// import useCPU from '../hooks/use-cpu';
+import useInterval from '../hooks/use-interval';
+
 
 import getRegisters from '../utils/registers';
 
-const RegisterPanel = ({ registers }) => {
+const RegisterPanel = () => {
+  const [cpuValid, setCpuValid] = useState(false);
+  const [cpu, setCpu] = useState(null);
+
+  useInterval(() => {
+    if (!cpuValid && window.myCpu) {
+      setCpuValid(true);
+    } else if (cpuValid) {
+      setCpu(window.myCpu);
+    }
+  }, 1000);
+
+  if (!cpuValid || !cpu) {
+    return <h1>Loading...</h1>;
+  }
+
+  console.log(cpu.instruction_amounts);
+
   const allRegisters = getRegisters();
 
-  const registersAppended = registers.map((reg, i) => (
+  const registersAppended = cpu.registers.map((reg, i) => (
     { ...allRegisters[i], value: reg.low_ }
```

Comment:
out of curiousity, what is reg.low_ and reg.hi_? like why is there a low and high?

Reply:
Im not sure tbh, we've been using low_ from the start, and thats the one that seems to carry the value. Ill look into this in case we're missing out on any data